internal/util/paramtable/global_param.go

```diff
@@ -1631,12 +1636,20 @@ func (p *dataCoordConfig) initMinioRootPath() {
 	p.MinioRootPath = rootPath
 }
 
+func (p *dataCoordConfig) initEnableAutoCompaction() {
+	p.EnableAutoCompaction = p.BaseParams.ParseBool("dataCoord.compaction.enableAutoCompaction", false)
+}
+
+func (p *dataCoordConfig) initEnableAutoExpiration() {
+	p.EnableAutoExpiration = p.BaseParams.ParseBool("dataCoord.compaction.enableAutoExpiration", false)
+}
+
 func (p *dataCoordConfig) initCompactionRetentionDuration() {
 	p.CompactionRetentionDuration = p.BaseParams.ParseInt64WithDefault("dataCoord.compaction.retentionDuration", 432000)
 }
 
-func (p *dataCoordConfig) initEnableAutoCompaction() {
-	p.EnableAutoCompaction = p.BaseParams.ParseBool("dataCoord.compaction.enableAutoCompaction", false)
+func (p *dataCoordConfig) initCompactionEntityExpiration() {
+	p.CompactionEntityExpiration = p.BaseParams.ParseInt64WithDefault("dataCoord.compaction.entityExpiration", 259200)
```

Comment:
just add a check here that CompactionEntityExpiration = math.max(CompactionEntityExpiration, CompactionRetentionDuration)

Reply:
There is no function Max for Int64 on Golang due to [this](https://www.gushiciku.cn/pl/2tBD).

So, I changed it like below
```
func (p *dataCoordConfig) initCompactionEntityExpiration() {
	p.CompactionEntityExpiration = p.BaseParams.ParseInt64WithDefault("dataCoord.compaction.entityExpiration", 259200)
	p.CompactionEntityExpiration = func(x, y int64) int64 {
		if x > y {
			return x
		}
		return y
	}(p.CompactionEntityExpiration, p.CompactionRetentionDuration)
}
```